### PR TITLE
Issue 10. Output beam and app under _build

### PR DESCRIPTION
**Pull request by cmkarlsson**
Compiled beam and app files now compiles to _build/{{profile}}/lib/{{app}}/ebin instead
of {{app}}/ebin

### CODE REVIEW

File: src/lr3_comp.erl [hunk 1]
```diff
@@ -50,7 +50,8 @@ compile_normal_app(AppInfo) ->
     AppDir       = rebar_app_info:dir(AppInfo),
     OtherSrcDirs = rebar_dir:src_dirs(Opts),
     SourceDirs   = lr3_comp_util:get_src_dirs(AppDir, ["src"] ++ OtherSrcDirs),
-    OutDir       = lr3_comp_util:relative_out_dir(AppInfo),
+    OutDir       = filename:join([rebar_app_info:out_dir(AppInfo),
+                                  "ebin"]),
```
Comment (yurrriq) [inline src/lr3_comp.erl]: Might it be worth rewriting (and renaming?) `lr3_comp_util:relative_out_dir/1`?

Comment (cmkarlsson) [inline src/lr3_comp.erl]: Actually I don't know why I did that. Looking at `lr3_com_util:relative_out_dir/1` is doing the exact same thing. 

I can change this line back and leave it as it was before.

As of rewriting. I think the compile plugin can be cut down in size depending on the scope of it. For example, is rebar3 lfe compile supposed to compile the entrie application or only lfe specific files? For example the app-file compilation is done by rebar3 anyway. And if it is working perhaps we should not change for the sake of change.

**Comment by yurrriq** [comment]
The application resource file is handle by `rebar3 compile` yeah, but what if a user has a pure LFE app and they don't call `rebar3 compile` at all? I think maybe a better approach would be too lazily handle it, i.e. deal with the APR iff it's not in the right place already.

**Comment by yurrriq** [comment]
I think we also have access to the rebar3 manifest so to speak, so maybe we could look there to determine if rebar3 already did or is planning to handle the application resource file for us. I can ask Tristan soon.

**Comment by cmkarlsson** [comment]
I don't have any particular preference, but not being stand-alone (you do depend on rebar3 anyway) can cut down code size and code duplication. 

If `rebar3 lfe compile` should be stand-alone it needs a little bit of love:)

One example is if you depend on `(include-lib "exemplar/include/html-macros.lfe")`. This doesn't compile with `rebar3 lfe compile` but works with `rebar3 compile`.

I'll open an issue for that.

**Comment by yurrriq** [comment]
True. Ok, I guess my preference would be to handle the application resource file lazily then.

**Comment by yurrriq** [comment]
@oubiwann, I think this is good to merge. One potential caveat is I can't remember when the automatic `-pa _build/default/lib/*/ebin` was added to LFE itself. This change could cause issues for older versions of LFE.

**Comment by yurrriq** [comment]
This has become high priority for me, as a user. I think I have merge powers now. I'll give this a good read through and test it today.

**Comment by yurrriq** [comment]
Thanks for the PR and for your patience, @cmkarlsson,

**Pull request merged**